tests/integration_tests/build/test_unittests.py

```diff
@@ -8,38 +8,42 @@
   is fixed
 """
 
-import os
-
-from subprocess import run
+import platform
 
 import pytest
 
-import host_tools.cargo_build as host  # pylint: disable=import-error
+import host_tools.cargo_build as host  # pylint:disable=import-error
 
+MACHINE = platform.machine()
+TARGETS = ["{}-unknown-linux-gnu".format(MACHINE),
+           "{}-unknown-linux-musl".format(MACHINE)]
 
-CARGO_UNITTEST_REL_PATH = os.path.join(host.CARGO_BUILD_REL_PATH, "test")
 
+@pytest.mark.parametrize(
+    "target",
+    TARGETS
+)
+def test_unittests(test_session_root_path, target):
+    """
+    Run unit and doc tests from all crates.
 
-@pytest.mark.timeout(240)
-def test_unittests(test_session_root_path):
-    """Run unit and doc tests from all crates using default target."""
-    run(
-        'CARGO_TARGET_DIR={} RUST_TEST_THREADS=1 RUST_BACKTRACE=1 cargo test '
-        '--all --no-fail-fast'.format(
-            os.path.join(test_session_root_path, CARGO_UNITTEST_REL_PATH),
-        ),
-        shell=True,
-        check=True
-    )
+    This will generate unit tests instances using the cartesian product of all
```

Comment:
The comment needs to be updated as it generates tests only for different targets.

Reply:
Done, take a look.